### PR TITLE
New overview for animations library

### LINKED ISSUE #34340: Animation library API docs need a general tutorial section
According to the survey the animation library's biggest problem is people just don't understand how everything fits together. We should create a long API doc section on the library's API docs showing how everything fits together, and then point to it from the various main classes and so on (maybe based on which ones get the most hits).

### PR DESCRIPTION
**Pull request by goderbauer**
## Description

Adds an overview of the animation library as a top-level doc comment to said library.

This has been pre-reviewed at https://docs.google.com/document/d/1-HDkRvlmgOwnNi4MmawPq6xoJjE5ANWfBakgPr1_JQY/edit.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/34340.

## Tests

I added the following tests:

* Doc only change, no tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/animation.dart [hunk 1]
```diff
@@ -6,9 +6,157 @@
...
+/// An [AnimationController] can be used wherever an [Animation] is expected. As
+/// the name implies, an [AnimationController] also provides control over its
+/// [Animation]: It implements methods to stop the animation at any time and to
+/// run it forward as well as in the reverse direction.
+///
+/// By default, an [AnimationController] increases its animation value linearly
+/// over the given duration from 0.0 to 1.0 when run in the forward direction.
+/// For many use cases you might want the value to be of a different type,
+/// change the range of the animation values, or change how the animation moves
+/// between values. This is achieved by wrapping the animation: An [Animatable]
```
Comment (Piinks) [inline packages/flutter/lib/animation.dart]: I'm not sure I understand this sentence with the colon. Do you wrap the animation in an Animatable?

File: packages/flutter/lib/animation.dart [hunk 2]
```diff
@@ -6,9 +6,157 @@
...
+/// at an animation value of 1.0 the status is "completed". When the animation
+/// triggers again to fade the widget back out, the animation status changes to
+/// "reverse" and the animation value runs back to 0.0. At that point the widget
+/// is fully faded out and the animation status switches back to "dismissed"
+/// until the animation is triggered again.
+///
+/// {@animation 420 100 https://flutter.github.io/assets-for-api-docs/assets/animation/animation_status_value.mp4}
+///
+/// Although you can't instantiate [Animation] directly (it is an abstract
+/// class) you can create one using an [AnimationController].
```
Comment (Piinks) [inline packages/flutter/lib/animation.dart]: Nit:
```suggestion
/// class), you can create one using an [AnimationController].
```

File: packages/flutter/lib/animation.dart [hunk 3]
```diff
@@ -6,9 +6,157 @@
...
+/// it could be the current background [Color] of a widget that transitions
+/// smoothly from one color to another. The current value of an animation is
+/// represented by an [Animation] object, which is the central class of the
+/// animation library. In addition to the current animation value, the
+/// [Animation] object also stores the current [AnimationStatus]. The status
+/// indicates whether the animation is currently conceptually running from the
+/// beginning to the end or the other way around. It may also indicate that the
+/// animation is currently stopped at the beginning or the end.
+///
+/// Other objects can register listeners on an [Animation] to be informed
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply packages/flutter/lib/animation.dart]: Good idea. Added links to addListener/addStatusListener.

File: packages/flutter/lib/animation.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/animation.dart]: Yes. Rephrased.

File: packages/flutter/lib/animation.dart (hunk 2 above)
Comment (goderbauer) [reply packages/flutter/lib/animation.dart]: Done.

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
LGTM! 🎉

**Comment by goderbauer** [comment]
Don't know why cirrus didn't update the status on this PR, all tests have passed on https://cirrus-ci.com/build/5717076037074944. Mergig.

**Pull request merged**